src/Evolution/Particles/MonteCarlo/GhostZoneCommunication.hpp

```diff
@@ -197,6 +197,94 @@ struct GhostDataMcPackets {
+// THIS FUNCTION IS CURRENTLY ONLY IMPLENTED IN 3D!
+// As other parts of the MC communication, it is also incompatible with
+// mesh refinement.
+template <size_t Dim>
+struct CombineCouplingDataPostStep {
+  using return_tags =
+      tmpl::list<Particles::MonteCarlo::Tags::CouplingTildeTau<DataVector>,
+                 Particles::MonteCarlo::Tags::CouplingTildeRhoYe<DataVector>,
+                 Particles::MonteCarlo::Tags::CouplingTildeS<DataVector, Dim>>;
+  using argument_tags =
+      tmpl::list<Particles::MonteCarlo::Tags::GhostZoneCouplingDataTag<Dim>,
+                 ::domain::Tags::Element<Dim>,
+                 evolution::dg::subcell::Tags::Mesh<Dim>>;
+  static void apply(
+      gsl::not_null<Scalar<DataVector>*> coupling_tilde_tau,
+      gsl::not_null<Scalar<DataVector>*> coupling_tilde_rho_ye,
+      gsl::not_null<tnsr::i<DataVector, Dim, Frame::Inertial>*>
+          coupling_tilde_s,
+      const Particles::MonteCarlo::GhostZoneCouplingData<Dim>& coupling_data,
+      const Element<Dim>& element, const Mesh<Dim>& subcell_mesh) {
+    // Check that we do not use this function for 1D/2D runs
+    ASSERT(Dim == 3, "CombineCouplingDataPostStep only coded in 3D so far");
+
+    const size_t num_ghost_zones = 1;
+    const Index<Dim> local_extents = subcell_mesh.extents();
+    Index<Dim> ghost_extents = local_extents;
+    for (size_t d = 0; d < 3; d++) {
+      ghost_extents[d] += 2 * num_ghost_zones;
+    }
+    // Loop over neighboring elements
+    for (const auto& [direction, neighbors_in_direction] :
+         element.neighbors()) {
+      for (const auto& neighbor : neighbors_in_direction) {
+        DirectionalId<Dim> directional_element_id{direction, neighbor};
+        if (coupling_data.coupling_tilde_tau.at(directional_element_id) ==
+            std::nullopt) {
+          continue;
+        }
+        const size_t dimension = directional_element_id.direction().dimension();
+        const Side side = directional_element_id.direction().side();
+        // Extents of coupling data: mesh with ghost points, sliced
+        // in direction 'dimension'
+        Index<Dim> ghost_zone_extents = ghost_extents;
+        ghost_zone_extents[dimension] = num_ghost_zones;
+        // These loops only work in 3D; needs fixing for other dimensions
+        for (size_t i = 0; i < ghost_zone_extents[0]; ++i) {
+          for (size_t j = 0; j < ghost_zone_extents[1]; ++j) {
+            for (size_t k = 0; k < ghost_zone_extents[2]; ++k) {
+              Index<Dim> index_3d{i, j, k};
+              // Index of point in ghost zone data
+              const size_t ghost_index =
+                  collapsed_index(index_3d, ghost_zone_extents);
+              // Index of corresponding live point in full
+              // mesh, with ghost zones.
+              index_3d[dimension] =
+                  (side == Side::Lower)
+                      ? index_3d[dimension] + num_ghost_zones
```

Comment:
why isn't this just `index_3d[dimension]`?

Reply:
I added a much longer comment. I assume that the difference with what you imagine is that we are copying from the ghost zones of the neighbor onto the live points of the current mesh (rather than the other way around, usually), and all meshes used here include ghost points in their indexing.

Reply:
ah, thanks, the comment helps a lot